examples/client/Locomotion/src/Components/PhoneNumberInput/index.js

```diff
@@ -4,65 +4,47 @@ import { AsYouType } from 'libphonenumber-js';
 import { getInputIsoCode } from '../../services/MccMnc';
 import i18n from '../../I18n';
 import { ERROR_COLOR } from '../../context/theme';
-import codes from './codes.json';
 
 const PhoneNumberInput = ({
   onPhoneNumberChange,
-  defaultCode,
   autoFocus,
   error,
   value,
 }) => {
   const [isFocused, setIsFocused] = useState(false);
-  const initialCode = codes.find(v => v.code === defaultCode);
-  const [prevCountryCode, setPrevCountryCode] = useState(initialCode.dialCode);
-  const [countryCode, setCountryCode] = useState(initialCode.dialCode);
-  const asYouTypePhoneNUmber = new AsYouType();
+  const [defaultCode, setDefaultCode] = useState(null);
+  const asYouTypePhoneNumber = new AsYouType();
 
   const onChangeText = (v) => {
-    const numberValue = `+${countryCode}${v}`;
-    asYouTypePhoneNUmber.input(numberValue);
+    const numberValue = `${v}`;
+    asYouTypePhoneNumber.input(numberValue);
+    const number = asYouTypePhoneNumber.getNumberValue();
+    console.log({number})
     return onPhoneNumberChange(
-      asYouTypePhoneNUmber.getNumberValue().replace('+', ''),
-      asYouTypePhoneNUmber.isValid(),
+      number && number.replace('+', ''),
+      asYouTypePhoneNumber.isValid(),
     );
   };
 
-  const changeCountryCode = (newCountryCode) => {
-    setPrevCountryCode(countryCode);
-    setCountryCode(newCountryCode);
-  };
-
-  const onChangeCountry = (v) => {
-    changeCountryCode(v.callingCode[0]);
-  };
-
-  useEffect(() => {
-    if (value) {
-      onChangeText(value.substring(prevCountryCode.length));
-    }
-  }, [countryCode]);
-
   const setIsoCode = async () => {
     const mobileIso = await getInputIsoCode();
-    changeCountryCode(mobileIso);
+    setDefaultCode(process.env.OVERWRITE_COUNTRY_CODE || mobileIso);
   };
 
   useEffect(() => {
     setIsoCode();
   }, []);
 
-  return (
+  return defaultCode ? (
     <PhoneInput
       value={value}
       autoFocus={autoFocus}
       defaultCode={defaultCode}
-      onChangeText={onChangeText}
+      onChangeFormattedText={onChangeText}
       textInputProps={{
         onFocus: () => setIsFocused(true),
         onBlur: () => setIsFocused(false),
       }}
-      onChangeCountry={onChangeCountry}
```

Comment:
we still need this functionality if the they dont want to use the default code no?

Reply:
onChangeFormattedText has both country code and actual number

Reply:
ah cool